### PR TITLE
PT-159406140 get latest channel state

### PR DESCRIPTION
**Pull request by cytadela8**
PT [SC WS: Get latest channel state](https://www.pivotaltracker.com/story/show/159406140)

### CODE REVIEW

File: apps/aehttp/src/sc_ws_api_jsonrpc.erl [hunk 1]
```diff
@@ -314,6 +314,15 @@ process_request(#{<<"method">> := <<"channels.get.balances">>,
         {error, _} ->
             {error, invalid_arguments}
     end;
+process_request(#{<<"method">> := <<"channels.get.offchain_state">>,
```
Comment (lucafavatella) [inline apps/aehttp/src/sc_ws_api_jsonrpc.erl]: Documentation at https://github.com/aeternity/protocol/blob/362d566af481214a5f650385b10d6cb3cfd0e831/node/api/channel_ws_api_json-rpc.md#examples may benefit from integration?
cc @uwiger

File: apps/aehttp/src/aehttp_api_encoder.erl [hunk 2]
```diff
@@ -204,7 +206,8 @@ type2pfx(peer_pubkey)      -> <<"pp">>;
 type2pfx(name)             -> <<"nm">>;
 type2pfx(state)            -> <<"st">>;
 type2pfx(poi)              -> <<"pi">>;
-type2pfx(state_trees)      -> <<"ss">>.
+type2pfx(state_trees)      -> <<"ss">>;
+type2pfx(call_state_tree)  -> <<"cs">>.
```
Comment (lucafavatella) [inline apps/aehttp/src/aehttp_api_encoder.erl]: It seems requiring integration of https://github.com/aeternity/protocol/blob/362d566af481214a5f650385b10d6cb3cfd0e831/node/api/api_encoding.md#encoding-scheme-for-api-identifiers-and-byte-arrays

**Comment by cytadela8** [comment]
@lucafavatella Thanks for links. I will create a aeternity/protocol PR.

**Comment by lucafavatella** [comment]
I noticed the CI had failed for CI issue - not PR issue. So I took the liberty of restarting failing jobs in workflow.

File: apps/aehttp/test/aehttp_integration_SUITE.erl [hunk 3]
```diff
@@ -3794,6 +3805,42 @@ query_balances_(ConnPid, Accounts, <<"json-rpc">>) ->
...
+decode_signed_tx(EncodedSignedTx) ->
+    {ok, SSignedTx} = aehttp_api_encoder:safe_decode(transaction, EncodedSignedTx),
+    aetx_sign:deserialize_from_binary(SSignedTx).
+
+query_state(ConnPid, Config) ->
+    query_state_(ConnPid, sc_ws_protocol(Config)).
+
+query_state_(ConnPid, <<"legacy">>) ->
+    %% get.offchain_state doesn't require any parameter, but not suplying one 
+    %% will remove "payload" key from request and cause error in legacy API
```
Comment (uwiger) [inline apps/aehttp/test/aehttp_integration_SUITE.erl]: Is this tracked in PT? It sounds like a bug.

Comment (cytadela8) [reply apps/aehttp/test/aehttp_integration_SUITE.erl]: This is a bug in testing framework, it only affects "legacy" protocol. As we are removing the "legacy" protocol we probably don't need to fix this bug.

**Pull request merged**